### DISCUSSION TITLE
Streamflow input data

### DISCUSSION THREAD

**Discussion by Mohammedhhu** [mhm-ufz/mHM#91] [Q&A] [answered]
Dear all 

I'm preparing to use mHM across the Blue Nile Basin; I have monthly data from 8 ground measurements of streamflow. So, can I put these data in a streamflow file and run the model on a monthly scale? or should the streamflow file be on a daily scale?

Thank you so much in advance 

Best regards,

Mohammed Abdallah, Sudan

**Comment by pallavhydro** [comment, marked as answer]
Hi @Mohammedhhu 

Streamflow data is not a model input in mHM. As long as you have the necessary input ready, you can run mhm. mHM internally runs at hourly time step, but its output can be saved at many temporal resolutions - sub-daily, daily, monthly, yearly (see the namelist files `mhm_output.nml`, `mrm_output.nml`)

Regarding the monthly streamflow data that you have, as you know, the role of streamflow measurements in hydrological modeling is to either evaluate the modelled streamflow or model optimisation. The issue with monthly streamflow data in mHM is that the temporal resolution of streamflow data has to be at least daily. mHM currently doesn't have monthly evaluation or optimisation. Following are some ideas that you can use as alternatives:

1. Create a daily time series such that your monthly data is allocated to a particular day of the month (day 1st day of each month) and keep the rest of the daily entries as missing data i.e. -9999. In this way, mHM can now use this "daily" time series (with lot of missing values) to evaluate or optimise the model. Please keep in mind that this method comes with inaccuracies, as the monthly streamflow data is usually the average streamflow over a month and not a particular day.

2. Another idea could be to run the model without using the streamflow data. Then, convert the daily simulations output of mhm (daily_discharge.out) to monthly values. Now, you can compare the monthly simulations to your monthly observations. Of course, in this method, you won't be able to optimise but only evaluate the model based on the parameter values you set in the namelist `mhm_parameters.nml`.

I hope this helps.
Pallav

**Comment by Mohammedhhu** [comment]
Hi Pallav

Thank you so much for your suggestion, I will follow it 

Best regards,,

Mohammed
